scan-payment/src/main/java/com/getbouncer/scan/payment/ml/ssd/SSD.kt

```diff
@@ -131,6 +145,13 @@ fun filterVerticalBoxes(detectedBoxes: List<DetectionBox>): List<DetectionBox> {
 
     val medianCenter = centers.elementAt(centers.size / 2)
     val medianHeight = heights.elementAt(heights.size / 2)
+    val aggregateDeviation = centers.map { abs(it - medianCenter) }.sum()
 
-    return detectedBoxes.filter { abs(it.rect.centerY() - medianCenter) <= medianHeight }
+    return if (aggregateDeviation > verticalOffset * medianHeight && detectedBoxes.size == QUICK_READ_LENGTH) {
+        return detectedBoxes.sortedBy { it.rect.centerY() }
```

Comment:
nit: you can omit the `return` here since the `if` statement above will return the result of this closure automatically